lua/nvim-treesitter-textobjects/swap.lua

```diff
@@ -6,20 +6,59 @@ local shared = require "nvim-treesitter-textobjects.shared"
 ---@field line integer
 ---@field character integer
 
----@param range1 TSTextObjects.Range
----@param range2 TSTextObjects.Range
+---@param range Range4
+---@return lsp.Range
+local function to_lsp_range(range)
+  ---@type integer, integer, integer, integer
+  local start_row, start_col, end_row, end_col = unpack(range)
+  return {
+    start = {
+      line = start_row,
+      character = start_col,
+    },
+    ["end"] = {
+      line = end_row,
+      character = end_col,
+    },
+  }
+end
+
+---@param range Range4
+---@return string[]
+local function get_text(bufnr, range)
```

Comment:
Can't this be replaced by `nvim_buf_get_text`?

Reply:
Done, and this might be another reason for keeping `Range4`, it is more intuitive for some APIs like this.

Reply:
Agreed.